ansible/Dockerfile

```diff
@@ -0,0 +1,705 @@
+ARG postgis_release=3.3.2
+ARG postgis_release_checksum=sha256:9a2a219da005a1730a39d1959a1c7cec619b1efb009b65be80ffc25bad299068
+ARG pgrouting_release=3.4.1
+ARG pgrouting_release_checksum=sha256:a4e034efee8cf67582b67033d9c3ff714a09d8f5425339624879df50aff3f642
+ARG pgtap_release=1.2.0
+ARG pgtap_release_checksum=sha256:9c7c3de67ea41638e14f06da5da57bac6f5bd03fea05c165a0ec862205a5c052
+ARG pg_cron_release=1.4.2
+ARG pg_cron_release_checksum=sha256:3652722ea98d94d8e27bf5e708dd7359f55a818a43550d046c5064c98876f1a8
+ARG pgaudit_release=1.7.0
+ARG pgaudit_release_checksum=sha256:8f4a73e451c88c567e516e6cba7dc1e23bc91686bb6f1f77f8f3126d428a8bd8
+ARG pgjwt_release=9742dab1b2f297ad3811120db7b21451bca2d3c9
+ARG pgsql_http_release=1.5.0
+ARG pgsql_http_release_checksum=sha256:43efc9e82afcd110f205b86b8d28d1355d39b6b134161e9661a33a1346818f5d
+ARG plpgsql_check_release=2.2.5
+ARG plpgsql_check_release_checksum=sha256:6c3a3c5faf3f9689425c6db8a6b20bf4cd5e7144a055e29538eae980c7232573
+ARG pg_safeupdate_release=1.4
+ARG pg_safeupdate_release_checksum=sha256:ff01d3d444d35924bd3d745c5695696292e2855042da4c30fe728fb3b6648122
+ARG timescaledb_release=2.9.1
+ARG timescaledb_release_checksum=sha256:883638f2e79d25ec88ee58f603f3c81c999b6364cb4c799919d363f04089b47b
+ARG wal2json_release=2_5
+ARG wal2json_release_checksum=sha256:b516653575541cf221b99cf3f8be9b6821f6dbcfc125675c85f35090f824f00e
+ARG pljava_release=1.6.4
+ARG pljava_release_checksum=sha256:5dc5a963365cbdeb786d0ad65b6310c4c5150ef8524e273bca4085c3fb0f056f
+ARG plv8_release=3.1.5
+ARG plv8_release_checksum=sha256:1e108d5df639e4c189e1c5bdfa2432a521c126ca89e7e5a969d46899ca7bf106
+ARG pg_plan_filter_release=5081a7b5cb890876e67d8e7486b6a64c38c9a492
+ARG pg_net_release=0.7
+ARG pg_net_release_checksum=sha256:f466d3de79b5fc9c4d0f137fcd6b51a868dacbf8002f94727d801553eb8e0c99
+ARG rum_release=1.3.13
+ARG rum_release_checksum=sha256:6ab370532c965568df6210bd844ac6ba649f53055e48243525b0b7e5c4d69a7d
+ARG pg_hashids_release=cd0e1b31d52b394a0df64079406a14a4f7387cd6
+ARG pg_graphql_release=1.1.0
+ARG pg_stat_monitor_release=1.1.1
+ARG pg_stat_monitor_release_checksum=sha256:1756a02d5a6dd66b892d15920257c69a17a67d48d3d4e2f189b681b83001ec2a
+ARG pg_jsonschema_release=0.1.4
+ARG vault_release=0.2.9
+ARG vault_release_checksum=sha256:1e813216395c59bb94c92be47ce8b70ba19ccc0efbcdb1fb14ed6d34a42c6cdb
+ARG groonga_release=12.0.8
+ARG groonga_release_checksum=sha256:7770c0ff6804ef4b47b015b15736cd973cffced977c20991b16b2daa4fea6eeb
+ARG pgroonga_release=2.4.0
+ARG pgroonga_release_checksum=sha256:5baaae0e7d81f8167e278e9a34c6ed56aece8b34f5ab98f228c64408093417b3
+ARG wrappers_release=0.1.7
+ARG hypopg_release=1.3.1
+ARG hypopg_release_checksum=sha256:e7f01ee0259dc1713f318a108f987663d60f3041948c2ada57a94b469565ca8e
+ARG pg_repack_release=1.4.8
+ARG pg_repack_release_checksum=sha256:18b4d871c1abf78cf0b1b1fe6081d435d183a8dc5eb977576e7a47fe113dd4ec
+ARG pgvector_release=0.4.0
```

Comment:
can we please avoid duplicating these version strings?

Reply:
These are default values which we can change via `--build-arg` flag. I will dedupe in a separate PR by passing in values from vars.yaml.

Currently there's a few inconsistencies in vars.yaml with using sha1 checksum and git branch ref. I will update everything to sha256 and commit ref in the follow up PR.

Reply:
Is there a reason we can't remove the defaults altogether? Can we update this PR with the vars.yml plumbing?

In general, I'm wary of the duplicated build process. Is there a path forward to getting rid of the duplication?

Reply:
> Is there a path forward to getting rid of the duplication?

One possible path forward to package each extension as a .deb file and publish to an internal registry / s3. Then it would be trivial (see pg_graphql for eg) for docker and ansible to install these packages separately from a single vars file. Wdyt?

Reply:
Another possibility to extract the docker image contents into a base VM. Fly.io uses this approach to bootstrap arbitrary firecracker VM.

Our ansible tasks can work on top of this base VM to do other configurations, like updating postgresql config.

Reply:
> Can we update this PR with the vars.yml plumbing?

I've created a new PR based on top of this to illustrate the plumbing we can do: https://github.com/supabase/postgres/pull/529

> Is there a reason we can't remove the defaults altogether?

These variables are currently inconsistent with the rest of extensions. Once we update them, should be ok to remove the defaults.
- `wal2json_commit_sha`: we probably want to rename as `wal2json_release` for consistency
- `pljava_release`/`pljava_release_checksum`: update sha256 after upstream PR merge
- `plv8_commit_version`: rename to `plv8_release`
- `pg_plan_filter_commit_version`: rename to `pg_plan_filter_release`
- `libsodium_release`/`libsodium_release_checksum`: remove after vault release
- `hypopg_commit_sha`: rename to `hypopg_release` and use release tag
- `pg_stat_monitor_release_checksum`/`pgvector_release_checksum`: currently missing from vars.yaml